### PR TITLE
Add Pulp profiling option

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Not to be merged until https://github.com/pulp/pulp/pull/2882 is merged and released in a Pulp release

**Comment by parthaa** [comment]
@ehelms do you want  to close http://projects.theforeman.org/issues/18617 or use this PR for that?

**Comment by ehelms** [comment]
This is passing tests now. @parthaa want to review it?

File: manifests/init.pp [hunk 1]
```diff
@@ -377,6 +382,8 @@
   $puppet_wsgi_processes     = $pulp::params::puppet_wsgi_processes,
   $migrate_db_timeout        = $pulp::params::migrate_db_timeout,
   $show_conf_diff            = $pulp::params::show_conf_diff,
+  $enable_profiling          = $pulp::params::enable_profiling,
+  $profiling_directory       = $pulp::params::profiling_directory,
```
Comment (stbenjam) [inline manifests/init.pp]: Would be good to have some validations on these, validate_bool validate_absolute_path respectively.

File: manifests/init.pp [hunk 2]
```diff
@@ -281,6 +281,11 @@
 #                               Defaults to false
 #                               type:boolean
 # 
+# $enable_profiling::           Turns on cProfiling of tasks in Pulp
+#                               type:boolean
+#
+# $profiling_directory::        Directory to store task profling data in
```
Comment (ekohl) [inline manifests/init.pp]: s/profling/profiling/

**Comment by ehelms** [comment]
@ekohl @stbenjam updated, please review

File: templates/server.conf.erb [hunk 3]
```diff
@@ -424,3 +424,24 @@ redirect_path: <%= scope['pulp::lazy_redirect_path'] %>
...
+#   This enables the cProfile Python module to profile individual tasks and
+#   store the output by task ID in a directory. Note that enabling this can
+#   impact performance.
+#
+# profile_directory:
+#   The directory that the cProfiles are stored in. This directory must be
+#   writeable and readable by Pulp.
+
+[profiling]
+<% unless [nil, :undefined, :undef, ''].include?(scope.lookupvar('pulp::enable_profiling')) -%>
```
Comment (ekohl) [inline templates/server.conf.erb]: We validate it's a boolean so this isn't needed.

File: templates/server.conf.erb [hunk 4]
```diff
@@ -424,3 +424,24 @@ redirect_path: <%= scope['pulp::lazy_redirect_path'] %>
...
+enable: <%= scope['pulp::enable_profiling'] %>
+<% end -%>
+<% unless [nil, :undefined, :undef, ''].include?(scope.lookupvar('pulp::profiling_directory')) -%>
```
Comment (ekohl) [inline templates/server.conf.erb]: We validate it's a valid path so this isn't needed.

**Comment by ehelms** [comment]
Updated and rebased

**Pull request merged**